optimum/habana/transformers/generation/utils.py

```diff
@@ -183,7 +183,7 @@ def _validate_model_kwargs(self, model_kwargs: Dict[str, Any]):
         if self.config.is_encoder_decoder:
             for key in ["decoder_input_ids"]:
                 model_kwargs.pop(key, None)
-        if self.config.model_type == "llama":
+        if self.config.model_type == "llama" or self.config.model_type == "falcon":
```

Comment:
Actually I removed the whole `_validate_model_kwargs` method as the issue with Llama has been fixed in Transformers v4.31 (I merged #312).
I tried Falcon without this and it worked on my side. Could you let me know if you get the same?

Reply:
Thanks for upgrading Transformers version, it worked on my side as well. Marking it resolved